src/java.desktop/windows/native/libawt/windows/awt_PrintJob.cpp

```diff
@@ -690,6 +691,7 @@ Java_sun_awt_windows_WPageDialogPeer__1show(JNIEnv *env, jobject peer)
             }
             ::GlobalUnlock(setup.hDevMode);
         }
+        doIt = JNI_TRUE;
     }
 
     AwtDialog::CheckUninstallModalHook();
```

Comment:
I wonder why the block of code at lines 697–709 is not needed if `JNI_FALSE` is returned as a result of an error condition.

No, it is not directly connected to the bug you're fixing, yet it doesn't look right to me.

Reply:
We can have a followup bug on this I guess since it was existing before [JDK-8307160](https://bugs.openjdk.org/browse/JDK-8307160) also..

Reply:
All I wanted is to bring up the inconsistency so that a few people would take a look at it while reviewing this change.

Reply:
It does look odd. Focus would need transferring in both cases I'd expect.
 It goes back to the very beginning of open source JDK so I can't see a changeset to point to in order to explain it. 
And I also can't find any bug reports that might be related - either one about adding it, or one about things not working because it is not always executed.

Reply:
I looked at the history before what's available in Git. I looks this has always been this way. Yet it doesn't look right.

`AwtDialog::CheckInstallModalHook()` is called right before the page dialog is displayed by using `::PageSetupDlg`. I expect `AwtDialog::CheckUninstallModalHook()` needs to be called after it.

Likely, the early returns (inside `if (ret)`) are very rare, if any of these has ever occurred at all.

I'll submit a bug to include calling `AwtDialog::CheckUninstallModalHook()` in error cleanup. The `done` label which were before [JDK-8307160](https://bugs.openjdk.org/browse/JDK-8307160) should've been before the line which calls `CheckUninstallModalHook`.